src/solver/clause.rs

```diff
@@ -230,6 +232,45 @@ impl Clause {
         )
     }
 
+    fn conditional_impl(
+        package_id: VariableId,
+        condition: VariableId,
+        then: Requirement,
+        candidates: impl IntoIterator<Item = VariableId>,
```

Comment:
Its unclear which candidates these are? Is it for the condition, the requirement or the combined set?